<!DOCTYPE html>
<html><head lang="en">
	<meta charset="utf-8" />
	<meta http-equiv="X-UA-Compatible" content="IE=edge"><title>OpenStack Nova error: libVirt cannot get CPU affinity of process 30619 - 世界的盡頭</title><meta name="viewport" content="width=device-width, initial-scale=1">
	<meta name="description" content="Nova-compute 無法運行，在 log 中發現:
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/eventlet/tpool.py&#34;, line 187, in doit
2015-02-11 15:34:05.511 30827 TRACE nova     result = proxy_call(self._autowrap, f, *args, **kwargs)
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/eventlet/tpool.py&#34;, line 147, in proxy_call
2015-02-11 15:34:05.511 30827 TRACE nova     rv = execute(f,*args,**kwargs)
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/eventlet/tpool.py&#34;, line 76, in tworker
2015-02-11 15:34:05.511 30827 TRACE nova     rv = meth(*args,**kwargs)
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/libvirt.py&#34;, line 2096, in vcpus
2015-02-11 15:34:05.511 30827 TRACE nova     if ret == -1: raise libvirtError (&#39;virDomainGetVcpus() failed&#39;, dom=self)
2015-02-11 15:34:05.511 30827 TRACE nova libvirtError: cannot get CPU affinity of process 30619: No such process
2015-02-11 15:34:05.511 30827 TRACE nova
" />
	<meta property="og:image" content=""/>
	<meta property="og:title" content="OpenStack Nova error: libVirt cannot get CPU affinity of process 30619" />
<meta property="og:description" content="Nova-compute 無法運行，在 log 中發現:
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/eventlet/tpool.py&#34;, line 187, in doit
2015-02-11 15:34:05.511 30827 TRACE nova     result = proxy_call(self._autowrap, f, *args, **kwargs)
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/eventlet/tpool.py&#34;, line 147, in proxy_call
2015-02-11 15:34:05.511 30827 TRACE nova     rv = execute(f,*args,**kwargs)
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/eventlet/tpool.py&#34;, line 76, in tworker
2015-02-11 15:34:05.511 30827 TRACE nova     rv = meth(*args,**kwargs)
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/libvirt.py&#34;, line 2096, in vcpus
2015-02-11 15:34:05.511 30827 TRACE nova     if ret == -1: raise libvirtError (&#39;virDomainGetVcpus() failed&#39;, dom=self)
2015-02-11 15:34:05.511 30827 TRACE nova libvirtError: cannot get CPU affinity of process 30619: No such process
2015-02-11 15:34:05.511 30827 TRACE nova
" />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://maple52046.github.io/posts/openstack-nova-error-libvirt-cannot-get-cpu-affinity-of-process/" /><meta property="og:image" content="https://maple52046.github.io/site-thumbnail.png" /><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2015-02-11T15:55:00+00:00" />
<meta property="article:modified_time" content="2015-02-11T15:55:00+00:00" />

<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:image" content="https://maple52046.github.io/site-thumbnail.png" /><meta name="twitter:title" content="OpenStack Nova error: libVirt cannot get CPU affinity of process 30619"/>
<meta name="twitter:description" content="Nova-compute 無法運行，在 log 中發現:
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/eventlet/tpool.py&#34;, line 187, in doit
2015-02-11 15:34:05.511 30827 TRACE nova     result = proxy_call(self._autowrap, f, *args, **kwargs)
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/eventlet/tpool.py&#34;, line 147, in proxy_call
2015-02-11 15:34:05.511 30827 TRACE nova     rv = execute(f,*args,**kwargs)
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/eventlet/tpool.py&#34;, line 76, in tworker
2015-02-11 15:34:05.511 30827 TRACE nova     rv = meth(*args,**kwargs)
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/libvirt.py&#34;, line 2096, in vcpus
2015-02-11 15:34:05.511 30827 TRACE nova     if ret == -1: raise libvirtError (&#39;virDomainGetVcpus() failed&#39;, dom=self)
2015-02-11 15:34:05.511 30827 TRACE nova libvirtError: cannot get CPU affinity of process 30619: No such process
2015-02-11 15:34:05.511 30827 TRACE nova
"/>

	
        <link href="https://maple52046.github.io/css/fonts.2c2227b81b1970a03e760aa2e6121cd01f87c88586803cbb282aa224720a765f.css" rel="stylesheet">
	

	
	<link rel="stylesheet" type="text/css" media="screen" href="https://maple52046.github.io/css/main.ac08a4c9714baa859217f92f051deb58df2938ec352b506df655005dcaf98cc0.css" />

	
	

	
	
	
	
</head>
<body>
        <div class="content"><header>
	<div class="main">
		<a href="https://maple52046.github.io">世界的盡頭</a>
	</div>
	<nav>
		
		
	</nav>
</header>

<main>
	<article>
		<div class="title">
			<h1 class="title">OpenStack Nova error: libVirt cannot get CPU affinity of process 30619</h1>
			<div class="meta">Posted on Feb 11, 2015</div>
		</div>
		

		<section class="body">
			<p>Nova-compute 無法運行，在 log 中發現:</p>
<pre tabindex="0"><code>2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/eventlet/tpool.py&#34;, line 187, in doit
2015-02-11 15:34:05.511 30827 TRACE nova     result = proxy_call(self._autowrap, f, *args, **kwargs)
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/eventlet/tpool.py&#34;, line 147, in proxy_call
2015-02-11 15:34:05.511 30827 TRACE nova     rv = execute(f,*args,**kwargs)
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/eventlet/tpool.py&#34;, line 76, in tworker
2015-02-11 15:34:05.511 30827 TRACE nova     rv = meth(*args,**kwargs)
2015-02-11 15:34:05.511 30827 TRACE nova   File &#34;/usr/lib/python2.7/dist-packages/libvirt.py&#34;, line 2096, in vcpus
2015-02-11 15:34:05.511 30827 TRACE nova     if ret == -1: raise libvirtError (&#39;virDomainGetVcpus() failed&#39;, dom=self)
2015-02-11 15:34:05.511 30827 TRACE nova libvirtError: cannot get CPU affinity of process 30619: No such process
2015-02-11 15:34:05.511 30827 TRACE nova
</code></pre><p>雖然看不太懂為什麼會出現這種錯誤，不過猜測大概與早上將 compute node 強制重開機有關。</p>
<p>看一下 <code>virsh list</code> 結果一大堆 VM 卡在 <em>running</em> 或 <em>shutdown</em> 的狀態。
但是並沒有任何 KVM process 在 running。</p>
<p>雖然不知道為什麼，但是我的直覺告訴我，只要解決 libVirt 這個狀態，就能解決 nova-compute 的問題</p>
<h1 id="solution">Solution</h1>
<p>找了一下，發現在 <strong>/var/run/libvirt/qemu/</strong> 這個資料夾中，有很多 instance-XXXXX.pid 與 instance-XXXXX.xml</p>
<pre tabindex="0"><code>root@compute-02: /var/run/libvirt/qemu# ls
instance-00000093.pid  instance-000002bb.xml  instance-00000307.pid  instance-00000327.xml  instance-00000340.pid
instance-00000093.xml  instance-000002be.pid  instance-00000307.xml  instance-0000032b.pid  instance-00000340.xml
instance-000000ac.pid  instance-000002be.xml  instance-00000322.pid  instance-0000032b.xml  instance-00000348.pid
instance-000000ac.xml  instance-000002c5.pid  instance-00000322.xml  instance-00000333.pid  instance-00000348.xml
instance-00000271.pid  instance-000002c5.xml  instance-00000325.pid  instance-00000333.xml  instance-00000357.pid
instance-00000271.xml  instance-000002c8.pid  instance-00000325.xml  instance-0000033d.pid  instance-00000357.xml
instance-000002bb.pid  instance-000002c8.xml  instance-00000327.pid  instance-0000033d.xml
</code></pre><p>這些檔案應該是要 instance running 時，libVirt 自己產生。我猜測 <code>virsh list</code> 應該是會來這個目錄讀取檔案。反正現在也沒有 VM 正在 running，乾脆把他們全部砍掉。</p>
<pre tabindex="0"><code>root@compute-02: /var/run/libvirt/qemu# rm -rf *
</code></pre><p>接下來重啟 libVirt ，然後就發現 <code>virsh list</code> 恢復到原本的狀態。</p>
<pre tabindex="0"><code>root@compute-02:~# service libvirt-bin restart
libvirt-bin stop/waiting
libvirt-bin start/running, process 29451
root@compute-02:~# virsh list
 Id    Name                           State
----------------------------------------------------

root@compute-02:~#
</code></pre><p>接著啟動 nova-compute 就可以恢復運作囉。</p>
<pre tabindex="0"><code>root@compute-02:~# service nova-compute start
nova-compute start/running, process 32664
root@compute-02:~# service nova-compute status
nova-compute start/running, process 32664
</code></pre>
		</section>

		<div class="post-tags">
			
			
			
		</div>
		</article>
</main>
<footer>
  <div style="display:flex"></div>
  <div class="footer-info">
    2024  <a
      href="https://github.com/athul/archie">Archie Theme</a> | Built with <a href="https://gohugo.io">Hugo</a>
  </div>
</footer>


<script>
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'G-GC1G5TJVGY', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>

</div>
    </body>
</html>
